docs/source/developers/cpp/building.rst

```diff
@@ -347,6 +352,50 @@ You can then invoke CMake to create the build directory and it will use the
 declared environment variable pointing to downloaded archives instead of
 downloading them (one for each build dir!).
 
+Statically Linking
+~~~~~~~~~~~~~~~~~~
+
+Static linking the Arrow libraries can be tricky depending on your build
```

Comment:
Is it tricky? Doesn't this PR handle most of the trickiness? As a user, it doesn't seem that tricky.

Reply:
"Tricky" is too much of an editorialization. It's accurate to say that it "may require an extra step beyond `find_project(Arrow)` in CMake". Will update